ensure_flutter_version.go

```diff
@@ -0,0 +1,224 @@
+	if installType.InstalledVersionsCommand != nil {
+		installsCmd := *installType.InstalledVersionsCommand()
+		if out, err := installsCmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+			return false, fmt.Errorf("list instances: %s", out)
+		} else if contains, err := f.containsVersion(out, required); err != nil {
+			return false, fmt.Errorf("check if installs contains version: %w", err)
+		} else if !contains {
+			return false, fmt.Errorf("version: %s channel: %s is not available in installed instances output: %s", required.version, required.channel, out)
+		} else {
+			f.Debugf("Version: %s channel: %s is available in installed instances output", required.version, required.channel)
+			return true, nil
+		}
+	}
+
+	return false, fmt.Errorf("no installed versions command defined for %s", installType.Name)
+}
+
+func (f *FlutterInstaller) containsVersion(output string, required flutterVersion) (bool, error) {
+	if output != "" {
+		versions, err := NewFlutterVersionList(output)
+		if err != nil {
+			return false, fmt.Errorf("parse releases: %w", err)
+		} else if len(versions) == 0 {
+			return false, fmt.Errorf("no versions available in releases output: %s", output)
+		} else {
+			for _, v := range versions {
+				if (required.version == "" || required.version == v.version) &&
+					(required.channel == "" || required.channel == v.channel) {
+					return true, nil
+				}
+			}
+		}
+	}
+
+	return false, fmt.Errorf("output is empty or doesnt contain version")
+}
+
+func (f *FlutterInstaller) ensureSetupFinished() error {
+	finsihSetupCmd := f.CmdFactory.Create("flutter", []string{"--version"}, nil)
+	f.Donef("$ %s", finsihSetupCmd.PrintableCommandArgs())
+	if out, err := finsihSetupCmd.RunAndReturnTrimmedCombinedOutput(); err != nil {
+		return fmt.Errorf("check flutter version after setup: %s %s", err, out)
+	} else {
+		f.Debugf("Flutter version output after install: %s", out)
+	}
+	return nil
+}
+
+func (f *FlutterInstaller) installAndSetDefault(installType *FlutterInstallType, required flutterVersion) error {
+	if installType.Install == nil {
+		return fmt.Errorf("no install command defined")
+	}
+
+	f.Debugf("Installing version: %s channel: %s with %s", required.version, required.channel, installType.Name)
+
+	if installType.ReleasesCommand != nil {
+		if hasRelease, err := f.hasRelease(installType, required); err != nil {
+			return fmt.Errorf("seaching for version in releases: %w", err)
+		} else if !hasRelease {
+			return fmt.Errorf("tool %s does not provide required version", installType.Name)
+		}
+	}
+
+	if err := installType.Install(required); err != nil {
+		return fmt.Errorf("install: %s", err)
+	} else if err := f.ensureSetupFinished(); err != nil {
+		f.Debugf("ensure setup is finished: %s", err)
+	}
+
+	if installType.SetDefault != nil {
+		if err := installType.SetDefault(required); err != nil {
+			return fmt.Errorf("set version default: %s", err)
+		} else if err := f.ensureSetupFinished(); err != nil {
```

Comment:
nit: we like to follow the offical Golang code conventions and write error handling `if`s slightly differently: https://go.dev/wiki/CodeReviewComments#indent-error-flow